client.go

```diff
@@ -41,48 +41,56 @@ type SingleServerClient struct {
 	isLocalTest bool
 }
 
-func NewClientImpl(deployFile string, targetName string, isLocalTest bool) (*SingleServerClient, error) {
+// Client which communicates with multiple underlying servers at once. Used if
+// the target is actually a group of servers
+type MultiServerClient struct {
```

Comment:
MultiTargetClient?  "...ServerClient" sounds a bit confused :P 

Or just call this Client, and rename Client to TargetClient or SingleTargetClient.  Otherwise you should rename the "NewClient" to something else.